.github/workflows/semgrep.yml

```diff
@@ -0,0 +1,22 @@
+name: Semgrep
+on:
+  push:
+     branches:
+          - varun-semgrep
+  #pull_request:
+  #workflow_dispatch:
```

Comment:
should this be changed to something like
```
branches:
  - master
pull_request:
```
This will allow the job to run on push to master and on every pull request, instead of just on this current branch (`varun-semgrep`).